vault/dynamic_system_view.go

```diff
@@ -88,6 +89,9 @@ func (d dynamicSystemView) SudoPrivilege(ctx context.Context, path string, token
 		return false
 	}
 
+	// AllowOperation path should be namespace scoped
+	path = strings.TrimPrefix(path, tokenNS.Path)
```

Comment:
I think this should be the `ctx`'s namespace path instead of `tokenNS.Path`.

Reply:
I think this is correct since `AllowedOperation` is already namespace-aware. However, I forgot to change the context to the context of the token. I fixed that.

Reply:
The path that you want to check against and trim would be the namespaced request path, similar to the logic in [your first commit in this PR](https://github.com/hashicorp/vault/pull/7224/commits/ac9d09cf942582128d1ce815081c7b86780c4791#diff-fe1fae01d64d9a473e37aa99564c5ea1R2106). Since `path` gets assigned back to `req.Path`, the trimmed prefix should be the namespace coming in from the request context.

There's also a helper in the namespace package, `ns.TrimmedPath()`, that makes this a bit more explicit (although it's essentially doing the same thing under the hood).

Reply:
Keep in mind that this isn't checking whether a request is allowed. It's checking whether a token's policies are allowed for a given path. This function could be called for a request that starts in `ns1/` but reference a path `ns2/foo/bar`, an entirely different namespace tree. The function is supposed to check whether the token's capabilities allow access to the given path, not whether it's part of the request path.

Reply:
Ah right, `path` in this case is not the request path, and `req` just is a dummy `logical.Request` that we pass in so that `AllowOperation` can check against it. Thanks for pointing this out!